[deliver] apple tv privacy policy update from deliver

aniketgaur opened the pull request:
<!-- Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->
it is the fix for this bug https://github.com/fastlane/fastlane/issues/14537
feature request is https://github.com/fastlane/fastlane/issues/14558

it also resolves this issue https://github.com/fastlane/fastlane/issues/14547 .


### Checklist
- [ ] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [ ] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [ ] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [ ] I've updated the documentation if necessary.

### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->

### Description
<!-- Describe your changes in detail -->
<!-- Please describe in detail how you tested your changes. -->

janpio commented on deliver/lib/deliver/upload_metadata.rb:
```diff
@@ -10,7 +10,7 @@ class UploadMetadata
 
     # Localised app details values
-    LOCALISED_APP_VALUES = [:name, :subtitle, :privacy_url]
+    LOCALISED_APP_VALUES = [:name, :subtitle, :privacy_url,:apple_tv_privacy_policy]
```
```suggestion
    LOCALISED_APP_VALUES = [:name, :subtitle, :privacy_url, :apple_tv_privacy_policy]
```

janpio commented:
Wow, that is all that is needed?

aniketgaur commented:
Yeah, I tried it worked.

janpio commented on deliver/lib/deliver/options.rb:
```diff
@@ -372,6 +372,10 @@ def self.available_options
                                      optional: true,
                                      is_string: false),
+        FastlaneCore::ConfigItem.new(key: :apple_tv_privacy_policy,
+                                     description: "Metadata: Localised apple tv privacy policy",
```
Let's make it more clear that this requires the text, and not an URL:
```suggestion
                                     description: "Metadata: Localised Apple TV privacy policy text",
```

giginet commented on deliver/lib/deliver/download_screenshots.rb:
```diff
@@ -13,7 +13,7 @@ def self.run(options, path)
 
     def self.download(options, folder_path)
-      v = options[:use_live_version] ? options[:app].live_version : options[:app].latest_version
+      v = options[:use_live_version] ? options[:app].live_version(platform: options[:platform]) : options[:app].latest_version(platform: options[:platform])
```
Could you explain to need this change?

giginet replied:
`platform` could be `nil`. https://github.com/fastlane/fastlane/blob/7da852267f38ab7e08ccc58b6cbc31d39dc6089f/deliver/lib/deliver/options.rb#L81
Do you need to care optional?

aniketgaur replied:
this was related to this issue https://github.com/fastlane/fastlane/issues/14547 .
when I am giving value for platform as appletvos it is not getting passed. So options[:app].latest_version(platform: options[:platform]) output "v" for platform "ios".

Will it break something else?

aniketgaur replied:
I have tested it. it works fine.

janpio replied:
@giginet meant that `options[:platform]` could be nil, thus giving `platform: nil` to `live_version` which might be a problem?

aniketgaur replied:
Default argument in live_version is platform:nil. If options [:platform] is nil then that will get pass. which was happening anyways before. Problem was when options [:platform] is not nil and it did not get pass in live_version method call.

aniketgaur commented:
I signed it!

aniketgaur commented:
@janpio and @giginet  what else is needed for the merge?

janpio commented:
#1: See https://github.com/fastlane/fastlane/pull/14569#issuecomment-483126311 (some of your commits are with different credentials)

aniketgaur commented:
both of them are mine. should I create new pr?

janpio commented:
You can also add the email address you used there to your CLA.
Or fix the commits and force push on your branch.
Or reset your branch, commit your changes in one swoop from the correct account, and then force push.
Or not force push, but do it all in a new branch and create a new PR for that one.

Commit:
apple tv privacy policy upload and download handling added
screenshot download and upload issue for platform appletvos when ios and appletvos both are present at appstoreconnect for same bundle id

aniketgaur commented:
@janpio  and @giginet  can you review it again?
 I have combined all commits in one.

joshdholtz reviewed (approved):
This is so 💯 Thanks for doing all of these fixes ❤️
